packages/overlay/src/integrations/sentry/components/insights/agents/AITraceDetails.tsx

```diff
@@ -0,0 +1,326 @@
+          </Table>
+        </div>
+      )}
+    </div>
+  );
+}
+
+function PromptSection({ trace }: { trace: SpotlightAITrace }) {
+  if (!trace.prompt) return null;
+
+  return (
+    <div>
+      <h2 className="mb-4 font-bold">Prompt</h2>
+      {trace.prompt.system && (
+        <div className="mb-4">
+          <h3 className="mb-2 text-sm font-semibold uppercase">System</h3>
+          <div className="bg-primary-900 p-2">
+            <pre className="whitespace-pre-wrap break-words font-mono text-sm">{trace.prompt.system}</pre>
+          </div>
+        </div>
+      )}
+
+      {trace.prompt.messages && trace.prompt.messages.length > 0 && (
+        <div>
+          <h3 className="mb-2 text-sm font-semibold uppercase">Messages</h3>
+          {trace.prompt.messages.map((message, i) => {
+            let messageContent: string;
+            if (message.role === 'assistant') {
+              if (!trace.response?.text) {
+                return null; // Don't render empty assistant messages
+              }
+              messageContent = trace.response.text;
+            } else {
+              messageContent = message.content;
+            }
+
+            return (
+              <div key={`message-${i}-${message.role}`} className="border-primary-700 mb-4 rounded border p-2">
+                <div className="mb-1 font-semibold capitalize">{message.role}</div>
+                <pre className="whitespace-pre-wrap break-words font-mono text-sm">{messageContent}</pre>
+              </div>
+            );
+          })}
+        </div>
+      )}
+    </div>
+  );
+}
+
+function ResponseSection({ trace }: { trace: SpotlightAITrace }) {
+  if (!trace.response) return null;
+
+  return (
+    <div>
+      <h2 className="mb-4 font-bold">Response</h2>
+      {trace.response.finishReason && (
+        <div className="mb-2">
+          <span className="text-sm font-semibold uppercase">Finish Reason: </span>
+          <span className="font-mono">{trace.response.finishReason}</span>
+        </div>
+      )}
+
+      {trace.response.text && (
+        <div className="mb-4">
+          <h3 className="mb-2 text-sm font-semibold uppercase">Assistant Message</h3>
+          <div className="border-primary-700 mb-4 rounded border p-2">
+            <div className="mb-1 font-semibold capitalize">assistant</div>
+            <pre className="whitespace-pre-wrap break-words font-mono text-sm">{trace.response.text}</pre>
+          </div>
+        </div>
+      )}
+    </div>
+  );
+}
+
+function ToolCallsSection({ trace }: { trace: SpotlightAITrace }) {
+  if (trace.toolCalls.length === 0) return null;
+
+  return (
+    <div>
+      <h2 className="mb-4 font-bold">Tool Calls</h2>
+      {trace.toolCalls.map((toolCall, i) => (
+        <ToolCallDetail key={`tool-${i}-${toolCall.toolCallId}`} toolCall={toolCall} />
+      ))}
+    </div>
+  );
+}
+
+export default function AITraceDetail() {
+  const { spanId } = useParams<{ spanId: string }>();
```

Comment:
Clarification:

An AI Trace is simply the span that meets a condition in a normal trace, along with all its descendants. So its `id` is the `Id` of that span